### ISSUE TITLE
"Produkt z takimi danymi już istnieje" - false positive

### ISSUE THREAD

**Issue by rasskabak** [allegro/allegro-api#10567]
Próbuję utworzyć nowy produkt w katalogu, z parametrami jak poniżej. W odpowiedzi dostaję "Produkt z takimi danymi już istnieje. Skontaktuj się z autorem aplikacji" i link do istniejącego produktu - https://api.allegro.pl/sale/products/8fa97328-a848-4fe1-ad94-6d271e050a28. Tyle, że ten produkt ma inny kod producenta i inny numer EAN.
Oba produkty to nić na cięciwy, przy czym w różnym kolorze. W tej kategorii nie ma parametru kolor, jednak po różnicy w kodzie EAN, kodzie producenta oraz modelu, nie powinny być one uznane jako ten sam produkt.

trace-id f846a880147467c5 
url: https://api.allegro.pl/sale/product-proposals
parametry:
Array
(
    [name] => Nić na cięciwę BCY B55 - 10m
    [category] => stdClass Object
        (
            [id] => 253941
        )

    [images] => Array
        (
            [0] => Array
                (
                    [url] => https://a.allegroimg.com/original/113a17/4cd6020a49819d8210ff97cb35ca
                )

        )

    [description] => stdClass Object
        (
            [sections] => Array
                (
                    [0] => stdClass Object
                        (
                            [items] => Array
                                (
                                    [0] => stdClass Object
                                        (
                                            [type] => TEXT
                                            [content] => <h1>Nić na cięciwę dacron BCY B55 - 10m</h1><p></p><ul><li><b>Kolor: Buckskin</b></li></ul>
                                        )

                                )

                        )

                    [1] => stdClass Object
                        (
                            [items] => Array
                                (
                                    [0] => stdClass Object
                                        (
                                            [type] => TEXT
                                            [content] => <p>Kod produktu: 2376</p>
                                        )

                                )

                        )

                )

        )

    [language] => pl-PL
    [parameters] => Array
        (
            [0] => stdClass Object
                (
                    [id] => 224017
                    [values] => Array
                        (
                            [0] => Strzaly.pl_2376_12120
                        )

                    [name] => Kod producenta
                )

            [1] => stdClass Object
                (
                    [id] => 237206
                    [values] => Array
                        (
                            [0] => Strzaly.pl_2376_12119
                        )

                    [name] => Model
                )

            [2] => stdClass Object
                (
                    [id] => 17448
                    [values] => Array
                        (
                            [0] => 0.003
                        )

                    [name] => Waga produktu z opakowaniem jednostkowym
                )

            [3] => stdClass Object
                (
                    [id] => 128975
                    [valuesIds] => Array
                        (
                            [0] => 128975_1161203
                        )

                    [name] => Marka
                )

            [4] => stdClass Object
                (
                    [id] => 238229
                    [valuesIds] => Array
                        (
                            [0] => 238229_413169
                        )

                    [name] => Rodzaj
                )

            [5] => stdClass Object
                (
                    [id] => 225693
                    [values] => Array
                        (
                            [0] => 5904172457837
                        )

                    [name] => EAN (GTIN)
                )
        )
)

**Comment by PrzemyslawLukanowski** [comment]
Podczas tworzenia produktów bierzemy pod uwagę różne reguły, które uwzględniają różne parametry, czasem również w połączeniu ze zdjęciami. Widzę, że. w tym przypadku najprawdopodobniej doszło do połączenia ze względu na podobieństwo zdjęcia, marki oraz modelu. Jeśli Twoim zdaniem powinniśmy inaczej rozpoznawać duplikaty w tej kategorii, zgłoś to proszę przez [formularz kontaktowy](https://allegro.pl/pomoc/kontakt).

**Comment by rasskabak** [comment]
W ogólności, kod EAN i/lub kod producenta, powinny jednoznacznie identyfikować produkt. Teraz mam np podobną sytuację:

trace-id: f13706ba179313e7

Sznurek w kolorze brązowym, w dwóch wariantach - z rdzeniem z żyłki oraz bez. Wszystkie parametry - długość, średnica, marka, kolor, typ i waga są takie same. Zdjęcia są identyczne. Natomiast różnią się zarówno kodem producenta jak i kodem EAN. Mimo tego, przy próbie dodania produktu do katalogu, dostaję komunikat, że produkt już istnieje.

I teoretycznie mógłbym nawet użyć tego produktu, który jest w katalogu, bo dla klienta nie zrobi to większej różnicy - szczegóły doczyta w opisie. Tyle, że jeśli użyję tego już istniejącego produktu, to zaraz inny Wasz skrypt uzna, że wystawiam powielone oferty **z tym samym produktem**, zakończy mi je, da ostrzeżenie, a jeśli sytuacja będzie się powtarzać - zablokuje mi konto

**Comment by rasskabak** [comment]
Oczywiście, wyślę zgłoszenie również przez wskazany formularz, ale z mojego dotychczasowego doświadczcenia wiem, że jeśli w takim zgłoszeniu pojawi się stwierdzenie, że "wystawiam oferty przez API", to zaraz wszyscy się wystraszą i odeślą mnie z powrotem tutaj -_-

**Comment by PrzemyslawLukanowski** [comment]
W tym przypadku wykrycie duplikatu nastąpiło na podstawie zdjęcia oraz parametrów długość i średnica. 

W zgłoszeniu możesz zaznaczyć, że nie ma znaczenia to, w jaki sposób wystawiasz oferty, ponieważ problem związany jest z regułami, na podstawie których identyfikujemy duplikaty produktu i że istniejące zasady powodują błędne wykrycia duplikatów. Zgłoszenie powinno zostać przekierowane do odpowiedniego zespołu związanego z katalogiem produktów, który przeanalizuje ten przypadek.
